lib/vicadmin/validate.go

```diff
@@ -209,14 +204,36 @@ func NewValidator(ctx context.Context, vch *config.VirtualContainerHostConfigSpe
 		v.DockerPort = fmt.Sprintf("%d", opts.DefaultTLSHTTPPort)
 	}
 
-	err = v.QueryDatastore(ctx, vch, sess)
+	err := v.QueryDatastore(ctx, vch, sess)
 	if err != nil {
 		log.Errorf("Had a problem querying the datastores: %s", err.Error())
 	}
 	v.QueryVCHStatus(vch, sess)
 	return v
 }
 
+func VCHName(ctx context.Context, sess *session.Session) string {
```

Comment:
OK so I have a few concerns about this function. For one, there's no comment string and it's named like a field so it's not clear what it does. Further, the log message at the end which states that it is "setting the VCH hostname" only makes sense in the context of its caller on line 116, where we _actually_ set the hostname. As written the function is a getter, but the log message claims it's a setter, and it's a function not a method so its name gives no hint as to which it is at all.

So! I would suggest transforming this function into a method which receives a `Validator` object, perform the assignment that is currently on line `116` inside this method instead, change the return type to `error`, and rename the method to `SetVCHName` and then on line 116 do: `if err = v.SetVCHName(); err != nil { /*handle error*/ } `